landbosse/landbosse_runner.py

```diff
@@ -0,0 +1,420 @@
+        )
+        if enable_cost_and_scaling_modifications:
+            xlsx_reader.apply_cost_and_scaling_modifications_to_project_parameters(
+                project_parameters
+            )
+
+        master_input_dict = xlsx_reader.create_master_input_dictionary(
+            data_sheets,
+            project_parameters,
+        )
+
+        output_dict: dict = {}
+        mc = Manager(input_dict=master_input_dict, output_dict=output_dict)
+        mc.execute_landbosse(project_name=project_parameters["Project ID with serial"])
+
+        # Extract result dataframes
+        runs_dict = {project_parameters["Project ID with serial"]: output_dict}
+        model_variables = pd.DataFrame(XlsxManagerRunner.extract_details_lists(None, runs_dict))
+        operation_cost = pd.DataFrame(
+            XlsxManagerRunner.extract_module_type_operation_lists(None, runs_dict),
+        )
+
+        # Save outputs into LandBOSSEResult object
+        self.result = LandBOSSEResult(
+            project_parameters=project_parameters,
+            data_sheets=data_sheets,
+            model_variables=model_variables,
+            operation_cost=operation_cost,
+        )
+
+    @staticmethod
+    def add_header_to_weather_dataframe(weather_window: pd.DataFrame) -> pd.DataFrame:
+        """Add blank header rows to a weather dataframe to match the format expected by
+         LandBOSSE.
+
+        Returns
+        -------
+        pd.DataFrame
+            Weather dataframe with blank header.
+        """
+        NUM_HEADER_ROWS = 5
+
+        weather_window = weather_window.reset_index()
+
+        # select columns required by LandBOSSE
+        weather_window = weather_window[
+            ["datetime", "surface_temperature", "surface_pressure", "wind_direction", "windspeed"]
+        ]
+
+        # Combine the weather data with blank header rows
+        weather_window = pd.concat(
+            objs=(
+                pd.DataFrame(index=range(NUM_HEADER_ROWS), columns=weather_window.columns).astype(
+                    weather_window.dtypes
+                ),
+                weather_window,
+            ),
+            ignore_index=True,
+        )
+        return weather_window
+
+    @staticmethod
+    def create_component_dataframe(
+        project_parameters: pd.Series,
+        data_sheets: dict[str, pd.DataFrame],
+    ) -> pd.DataFrame:
+        """Convert the data from the input component data dictionary to the dataframe format
+        required by LandBOSSE. This includes rows for each separate blade and tower section.
+
+        Parameters
+        ----------
+        project_parameters : pd.Series
+            LandBOSSE YAML inputs
+        data_sheets : dict[str, pd.DataFrame]
+            LandBOSSE excel input tables
+
+        Returns
+        -------
+        pd.DataFrame
+            Component data in LandBOSSE format
+        """
+        NUM_BLADES = 3
+
+        component_param = project_parameters.pop("component")
```

Comment:
Should the data be being removed from `proejct_parameters`? Even though this is happening inside this method, it will be directly modifying whatever is passed to `project_parameters` and want to be sure that's the intended behavior.